contracts/test/proposals/DeployProposal.s.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.13;
+
+import "hardhat/console.sol";
+
+import {vip16 as vip} from "./vips/vip16.sol";
+import {Script} from "../../../forge-std/src/Script.sol";
+import {Addresses} from "./Addresses.sol";
+
+/*
+How to use:
+1/ Update the PRIVATE_KEY variable in this script to read the proper env variable
+2/ Configure DO_DEPLOY, DO_AFTERDEPLOY, DO_TEARDOWN in this file as needed
+2/ Import and inherit the proper VIP proposal script
+3/ Run the following command, with correct RPC url :
+forge script contracts/test/proposals/DeployProposal.s.sol:DeployProposal \
+    -vvvv \
+    --rpc-url $LOCAL_RPC_URL \
+    --broadcast
+Remove --broadcast if you want to try locally first, without paying any gas.
+*/
+
+contract DeployProposal is Script, vip {
+    uint256 public PRIVATE_KEY = vm.envUint("ANVIL0_PRIVATE_KEY");
+    bool public DO_DEPLOY = true;
+    bool public DO_AFTERDEPLOY = true;
+    bool public DO_TEARDOWN = false;
```

Comment:
shouldn't these be set from environment variables?

Reply:
that's doable but then the script don't run if we don't define these environment variables

Reply:
It's confusing to me having these variables set the way they are when it means the code can only run one path based on how they're set. Is the idea to have individuals toggle flags in this file based on the behavior they need? If so, we should move to env variables, if not, then we can delete these variables and not create additional branches with the if statements.

Reply:
we could have default values, that starts out in their current value, and then if it reads in a string from the environment that says something different, use that instead.